closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityProperties.java

```diff
@@ -0,0 +1,765 @@
+                            .printStackTrace();
+                    System.exit(1);
+                }
+
+                // Constraint object array.
+                // For each constraint type, algorithm and attributes.
+                Constraint[] constraints = new Constraint[constrArray.length];
+
+                int cNum = 0;
+                for (String constraint : constrArray) {
+                    String[] input = constraint.split(",");
+
+                    // Constraint must includes 3 fields(type, algorithm, attributes)
+                    if (input.length != 3) {
+                        new RuntimeException(
+                                "Constraint format is incorrect: " + providerInfo)
+                                .printStackTrace();
+                        System.exit(1);
+                    }
+
+                    constraints[cNum] = new Constraint(input[0].trim(), input[1].trim(), input[2].trim());
+
+                    if (debug != null) {
+                        debug.println("Loading constraints for provider " + providerName
+                                + " with constraints type: " + constraints[cNum].getType()
+                                + " algorithm: " + constraints[cNum].getAlgorithm()
+                                + " attributes: " + constraints[cNum].getAttributes());
+                    }
+                    cNum++;
+                }
+                providerConstraints.put(providerName, constraints);
+                if (debug != null) {
+                    debug.println("Loaded constraints for security provider: " + providerName);
+                }
+            } else {
+                new RuntimeException(
+                        "Constraint format is incorrect: " + providerInfo)
+                        .printStackTrace();
+                System.exit(1);
+            }
+        }
+    }
+
+    /**
+     * Check if the Service is allowed in restricted security mode.
+     *
+     * @param service the Service to check
+     * @return true if the Service is allowed
+     */
+    public boolean isServiceAllowed(Service service) {
+
+        String providerName = service.getProvider().getName();
+        String type = service.getType();
+        String algorithm = service.getAlgorithm();
+
+        // Provider with argument, remove argument.
+        // e.g. SunPKCS11-NSS-FIPS, remove argument -NSS-FIPS.
+        int pos = providerName.indexOf('-');
+        providerName = (pos < 0) ? providerName : providerName.substring(0, pos);
+
+        Constraint[] constraints = providerConstraints.get(providerName);
+
+        // Go into the security provider constraints check if there are.
+        if (constraints != null && constraints.length > 0) {
+
+            for (Constraint constraint : constraints) {
+
+                boolean cTypePut = "*".equals(constraint.getType())
+                        || type.equals(constraint.getType());
```

Comment:
Repeated subexpressions (`constraint.getType()` in this case); suggest a helper method that checks for `"*"` or a specific string.